streams/src/main/java/org/apache/kafka/streams/processor/internals/PartitionGroup.java

```diff
@@ -266,6 +270,10 @@ StampedRecord nextRecord(final RecordInfo info, final long wallClockTime) {
             if (record != null) {
                 --totalBuffered;
 
+                totalBytesBuffered -= (record.key() != null ? record.serializedKeySize() : 0) +
```

Comment:
How about adding a `sizeInBytes` function to `StampedRecord` class, to avoid duplicating the calculation each time?

Also we do not need to check for `key/value == null` or not, since in that case the `ConsumerRecord#serializedKeySize/ValueSize` would be 0.

Also note that the total bytes taken here is more than just the key/value since we have other fields like theaders, timestamps etc, but since it's hard to capture exactly how many bytes are taken in JVM to store a ConsumerRecord<byte[], byte[]> I think getting just getting the sum of key/value/timestamp/offset/topic/partition/headers is fine.